woocommerce/woocommerce
# Add coming soon template - modern black

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #52443 

This PR adds a new Coming Soon template- modern black, built using WP blocks. The template follows the Figma design and ensures that all elements, including images and copy, are fully editable to provide flexibility for customization.

Figma: 0BWMerqRR1f468EwC7g5tg-fi-3655_107812





### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site with  `coming-soon-newsletter-template` feature flag enabled.
2. Enable coming soon mode in site visibility settings.
3. Go to `Appearance > Editor > Templates > Coming soon`
4. Choose `Coming Soon Modern Black` template on the right sidebar (Template > Design > Coming Soon Modern Black)
6. Confirm the template matches the design in Figma.
7. Confirm that you can customize the styles, replace image, change the content, etc.
8. Select `Coming Soon Modern Black` template again and save the template.
9. Go to the site front e incognito tab to make sure the template is loaded correctly. The page should be responsive.
10. Go to `Appearance > Editor > Templates` and confirm the coming soon template preview shows the selected template.

Figam Design:

![Coming soon template - Login](https://github.com/user-attachments/assets/463a2019-4e83-4e31-ac66-36ba5707ad8b)

Screenshots:

![Screenshot 2024-11-25 at 13 43 51](https://github.com/user-attachments/assets/19559bd1-34dd-4ac2-97c1-2432ba090cdb)

![Screenshot 2024-11-25 at 13 45 52](https://github.com/user-attachments/assets/f35a6641-6701-47c5-a41a-5cb1acbf19ed)

![Screenshot 2024-11-25 at 13 46 29](https://github.com/user-attachments/assets/c0de765e-8dbe-409a-aa29-9b3501fc740e)


It's expected that the template preview doesn't fill the container height because the ratio and the element height don't match. This is similar to 404 template but because the background color is dark, it's more obvious.


![Screenshot 2024-11-25 at 13 57 28](https://github.com/user-attachments/assets/5cd77195-b761-4e5f-bbb6-e0292263cfc6)




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
